packages/dataviews/src/view-list.js

```diff
@@ -1,9 +1,70 @@
 /**
- * Internal dependencies
+ * WordPress dependencies
  */
-import ViewTable from './view-table';
+import { useAsyncList } from '@wordpress/compose';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
 
-export default function ViewList( props ) {
-	// To do: change to email-like preview list.
-	return <ViewTable { ...props } />;
+export default function ViewList( {
+	view,
+	fields,
+	data,
+	getItemId,
+	onSelectionChange,
+} ) {
+	const shownData = useAsyncList( data, { step: 3 } );
+	const mediaField = fields.find(
+		( field ) => field.id === view.layout.mediaField
+	);
+	const primaryField = fields.find(
+		( field ) => field.id === view.layout.primaryField
+	);
+	const visibleFields = fields.filter(
+		( field ) =>
+			! view.hiddenFields.includes( field.id ) &&
+			! [ view.layout.primaryField, view.layout.mediaField ].includes(
+				field.id
+			)
+	);
+
+	return (
+		<ul className="dataviews-list-view">
+			{ shownData.map( ( item, index ) => {
+				return (
+					// TODO: make li interactive.
```

Comment:
@andrewhayward what would be the recommended way to make this work? Either making `li` interactive or substituting it by something else?

Reply:
I went with https://github.com/WordPress/gutenberg/pull/56746/commits/d0aaf512bef7a5fd49f86650e1d7cb94f79687bf.

(It's not shown in the video but I pressed the `ENTER` key to trigger the preview):

https://github.com/WordPress/gutenberg/assets/583546/28973d9d-1660-4226-be14-42a49d4c4048